enclave-tls/samples/enclave-tls-client/client.c

```diff
@@ -38,20 +38,49 @@ static sgx_enclave_id_t load_enclave(bool debug_enclave)
 	int updated = 0;
 	int ret = sgx_create_enclave(ENCLAVE_FILENAME, debug_enclave, &t, &updated, &eid, NULL);
 	if (ret != SGX_SUCCESS) {
-		ETLS_ERR("Failed to load enclave %d\n", ret);
+		printf("Failed to load enclave %d\n", ret);
 		return -1;
 	}
 
-	ETLS_DEBUG("Success to load enclave id %ld\n", eid);
+	printf("Success to load enclave id %ld\n", eid);
 
-	return eid;
+	return (int64_t)eid;
 }
-#endif
 // clang-format on
 
-int enclave_tls_echo(int fd, enclave_tls_log_level_t log_level, char *attester_type,
-		     char *verifier_type, char *tls_type, char *crypto_type, bool mutual,
-		     bool debug_enclave)
+int enclave_tls_client_startup(enclave_tls_log_level_t log_level, char *attester_type,
+			       char *verifier_type, char *tls_type, char *crypto_type, bool mutual,
+			       bool debug_enclave, char *ip, int port)
+{
+	int64_t enclave_id = 0;
+	unsigned long flags = 0;
+	uint32_t s_ip = inet_addr(ip);
+	uint16_t s_port = htons((uint16_t)port);
+
+	enclave_id = load_enclave(debug_enclave);
+	if (enclave_id == -1) {
+		printf("Failed to load sgx stub enclave\n");
+		return -1;
+	}
+	if (mutual)
+		flags |= ENCLAVE_TLS_CONF_FLAGS_MUTUAL;
+
+	int ret = 0;
+	int sgx_status = ecall_etls_client_startup((sgx_enclave_id_t)enclave_id, &ret,
+						   (sgx_enclave_id_t)enclave_id, log_level,
+						   attester_type, verifier_type, tls_type,
+						   crypto_type, flags, s_ip, s_port);
+	if (sgx_status != SGX_SUCCESS || ret)
+		printf("failed to startup client: sgx status %#x return %#x\n", sgx_status, ret);
+
+	return ret;
+}
+#endif
+
+#ifndef SGX
+int enclave_tls_client_startup(enclave_tls_log_level_t log_level, char *attester_type,
+			       char *verifier_type, char *tls_type, char *crypto_type, bool mutual,
+			       bool debug_enclave, char *ip, int port)
```

Comment:
Not sure what happens here. I just see you are doing:

```C
#ifdef OCCLUM
...
#elif defined(SGX)
int enclave_tls_client_startup() {
   ...
}
#endif

#ifndef SGX
int enclave_tls_client_startup() {
   ...
}
#endif
```

Could you improve the readability? Such as
```C
#ifdef OCCLUM
...
#elif defined(SGX)
...
#endif // for header only

#ifdef SGX
int enclave_tls_client_startup() {
   ...
}
#else
int enclave_tls_client_startup() {
   ...
}
#endif
```